code_folder/A1_csv_to_json.py

```diff
@@ -0,0 +1,356 @@
+import pandas as pd
+import os
+import json
+from pathlib import Path
+import logging
+
+"""
+How to use:
+all default input csv"s are stored in "/mvs_eland/inputs/elements/default_csv".
+These csvs are not to be changed! All csvs that you need in order to set up 
+your energy system should be stored into "/mvs_eland/inputs/elements/csv". Here
+you can change parameters and add components. The given parameters need to be 
+maintained. 
+Storage: The "energyStorage.csv" contains information about all storages. 
+For each storage there needs to be another file named exactly as the according 
+storage-column in "energyStorage.csv", usually this is "storage_01", 
+"storage_02" etc. Please stick to this convention.
+After all the function "infer_resources()" reads all csv that are stored in 
+the folder "/mvs_eland/inputs/elements/csv" and creates one json input file 
+for mvs
+"""
+
+
+def infer_resources():
+
+    """
+    Method looks at all csv-files in "mvs_eland/inputs/elements/csv",
+    converts them into json files and joins them together into the file
+    "mvs_eland/inputs/working_example2.json". When reading the csv files it is
+    checked, weather all required parameters for each component are given.
+    Missing parameters will return a message.
+
+    Parameters
+    -------------
+    None
+
+    Returns
+    --------------
+    None
+        the json file "working_example2.json" is saved
+
+    """
+    input_directory = os.path.join(
+        Path(os.path.dirname(__file__)).parent, "inputs/elements/"
+    )
+    output_filename = "working_example2.json"
+
+    create_input_json(input_directory, output_filename)
+
+
+def create_input_json(input_directory, output_filename, pass_back=False):
+
+    """
+    reads all csv files in "mvs_eland/inputs/elements/csv", checks if all
+    required parameters for each csv file are available and saves them into
+    one json file
+
+    :param input_directory: str
+        name of the input directory where the csv"s can be found
+    :param output_filename: str
+        name of the output file with ending
+    :param pass_back: binary
+        if pass_back=True: the final json dict is returned. Otherwise it is
+        only saved
+
+    :return: None
+        saves
+    """
+    logging.info(
+        "loading and converting all csv's from /mvs_eland/inputs/elements/csv"
+        " into one json"
+    )
+    input_json = {}
+    for f in os.listdir(os.path.join(input_directory, "csv/")):
+
+        if f == "energyConsumption.csv":
+            parameters = ["dsm", "file_name", "label", "type_asset", "type_oemof"]
```

Comment:
Add comment:
# Hardcoded parameter list of energyConsumption assets

...analogeously for other csv's

Reply:
done